test/integration/targets/hwc_vpc_route/tasks/main.yml

```diff
@@ -0,0 +1,117 @@
+---
+# ----------------------------------------------------------------------------
+#
+#     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+#
+# ----------------------------------------------------------------------------
+#
+#     This file is automatically generated by Magic Modules and manual
+#     changes will be clobbered when the file is regenerated.
+#
+#     Please read more about how to change this file at
+#     https://www.github.com/huaweicloud/magic-modules
+#
+# ----------------------------------------------------------------------------
+# Pre-test setup
+- name: create a vpc
+  hwc_network_vpc:
+    cidr: "192.168.0.0/16"
+    name: "ansible_network_vpc_test_local"
+    state: present
+  register: vpc1
+- name: create a vpc
+  hwc_network_vpc:
+    cidr: "192.168.0.0/16"
+    name: "ansible_network_vpc_test_peering"
+    state: present
+  register: vpc2
+- name: create a peering connect
+  hwc_vpc_peering_connect:
+    local_vpc_id: "{{ vpc1.id }}"
+    name: "ansible_network_peering_test"
+    filters:
+      - "name"
+    peering_vpc:
+      vpc_id: "{{ vpc2.id }}"
+    state: present
+  register: connect
+- name: delete a route
+  hwc_vpc_route:
+    vpc_id: "{{ vpc1.id }}"
+    destination: "192.168.0.0/16"
+    next_hop: "{{ connect.id }}"
+    state: absent
+#----------------------------------------------------------
+- name: create a route
+  hwc_vpc_route:
+    vpc_id: "{{ vpc1.id }}"
+    destination: "192.168.0.0/16"
+    next_hop: "{{ connect.id }}"
+    state: present
+  register: result
```

Comment:
For every test, you should first do it with `check_mode: yes`. Also, you should do idempotence checks (also with check mode).

Reply:
Added check_mode test and idempotence with check mode test.